### PR TITLE
Ruler HA - Proposal

### PR DESCRIPTION
**Pull request by rajagopalanand**
**What this PR does**:

**Which issue(s) this PR fixes**:

Proposal for Ruler HA

I can add more details as necessary

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/proposals/ruler-ha-new.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+This redundancy will allow for missed rule evaluations from single ruler outages to be covered by other instances evaluating the same rule groups.
+
+To avoid inconsistent rule group state, which is maintained by Prometheus, the author proposes making a change in Prometheus rule group evaluation logic as described below
+
+### Prometheus change
+
+The author proposes making a change to Prometheus to allow for pausing and resuming (or activating and deactivating) a rule group as described [here](https://github.com/prometheus/prometheus/issues/13630)
+
+If the proposal is not accepted by Prometheus community, the proposal is to maintain a fork of Prometheus for Cortex with modified rule group evaluation behavior. This [draft PR](https://github.com/prometheus/prometheus/pull/13858)
```
Comment (friedrichg) [inline docs/proposals/ruler-ha-new.md:41]: >maintain a fork of Prometheus for Cortex

Maintaining a full prometheus fork should be the last option, we don't have the capacity to maintain such fork. I think we could instead copy some of the prometheus ruler code here and call it experimental. do you think that could work or is that not a possibility?

**Review by friedrichg — COMMENTED** [review]
Thanks for the spec.

Run make clean-white-noise to pass the lint

File: docs/proposals/ruler-ha-new.md (hunk 1 above)
Comment (rajagopalanand) [reply docs/proposals/ruler-ha-new.md:41]: I believe this is doable. Probably need to copy all the ruler code from [this package](https://github.com/prometheus/prometheus/tree/main/rules)

**Comment by yeya24** [comment]
Let's update the proposal with your latest proposed changes? IMO this should be merged before #6129.
@rajagopalanand @rapphil

**Review by yeya24 — COMMENTED** [review]
I see we use the same Title `Ruler HA` as the old proposal https://cortexmetrics.io/docs/proposals/ruler-ha/.

We changed the file name but I think the title will be the same when looking at https://cortexmetrics.io/docs/proposals/.
Should we only keep one proposal visible? Or let's change the title for this proposal or the old proposal

**Review by rapphil — APPROVED** [review]

**Review by yeya24 — APPROVED** [review]

**Pull request merged**